### PR TITLE
Keep segment statslog if no insert applied

### PR DESCRIPTION
**Pull request by congqixia**
Related to #18220 #19610

/kind improvement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/datanode/segment.go [hunk 1]
```diff
@@ -45,9 +46,10 @@ type Segment struct {
 	startPos *internalpb.MsgPosition // TODO readonly
 	endPos   *internalpb.MsgPosition
 
-	pkFilter *bloom.BloomFilter //  bloom filter of pk inside a segment
-	minPK    primaryKey         //	minimal pk value, shortcut for checking whether a pk is inside this segment
-	maxPK    primaryKey         //  maximal pk value, same above
+	statsChanged bool               //  segment stats changed
```
Comment (jiaoew1991) [inline internal/datanode/segment.go:49]: this variable maintained in `Segment` looks bad

Comment (congqixia) [reply internal/datanode/segment.go]: Fixed

**Pull request merged**